server/src/main/java/org/opensearch/rest/BaseRestHandler.java

```diff
@@ -122,10 +122,6 @@ public final void handleRequest(RestRequest request, RestChannel channel, NodeCl
             throw new IllegalArgumentException(unrecognized(request, unconsumedParams, candidateParams, "parameter"));
         }
 
-        if (request.hasContent() && request.isContentConsumed() == false) {
-            throw new IllegalArgumentException("request [" + request.method() + " " + request.path() + "] does not support having a body");
-        }
-
```

Comment:
Do we also have to remove this check from [other files](https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch%20request.hasContent()&type=code) as well?

Reply:
I don't see any other references `request.hasContent()` still seems valid use case to me.

Reply:
It's possible some plugins use those methods so I deliberately chose not to revert the entire commit and just "intentionally ignore behavior that was already ignored accidentally".   Devs should still be allowed to enforce body-related checks in their plugins if they want, they just shouldn't expect OpenSearch to do it for them.